private/buf/cmd/buf/command/curl/curl.go

```diff
@@ -916,6 +955,9 @@ func run(ctx context.Context, container appext.Container, f *flags) (err error)
 			// This shouldn't be possible since we check in flags.validate, but just in case
 			return nil, errors.New("URL positional argument is missing")
 		}
+		if f.HTTP3 {
+			return makeHTTP3Client(f, bufcurl.GetAuthority(host, requestHeaders), container.VerbosePrinter())
+		}
```

Comment:
Maybe instead have rename` makeHTTPClient` -> `makeHTTPRoundTripper`, and have it delegate to `makeHTTP3RoundTripper` internally. Then you can factor the two calls to `bufcurl.NewVerboseHTTPClient` into a single call right here.

Reply:
I believe I've implemented your suggestion. I agree, it looks a bit better!